src/github-url.js

```diff
@@ -0,0 +1,137 @@
+require('./config');
+const parseGithubUrl = require('parse-github-url');
+const fetch = require('node-fetch');
+
+const githubAPI = 'http://api.github.com';
+
+
+const isValidGithubUrl = (ghUrl) => ghUrl.protocol === 'https:'
+                                  && ghUrl.host === 'github.com'
+                                  && (ghUrl.branch !== 'issues'
+                                    || ghUrl.branch !== 'pull'
+                                    || ghUrl.branch !== 'master');
+
+/**
+ * @description Takes a url, checks if it's a valid github url
+ * and returns JSON with fetched data related to the url from Github's V3 API.
+ * At the moment, it accepts URLs of repositories, issues and pull requests.
+ * @link https://developer.github.com/v3/
+ * @param {string} url Url that will be processed.
+ * @return An object with all the fetched data.
+ */
+exports.getGithubUrlData = async (incomingUrl) => {
+  const ghUrl = parseGithubUrl(incomingUrl);
```

Comment:
in here, you might want to wait for the `parseGithubUrl` to get the job done, and then check `isValidGithubUrl`.
something like:
```
const ghUrl = await parseGithubUrl(incomingUrl);
```

Reply:
Very well done. Very nice, clean and properly commented code. I have learned always from your code.

Reply:
@Reza-Rajabi from the [parse-github-url ](https://github.com/jonschlinkert/parse-github-url/blob/master/index.js) documentation, it doesn't look like the function returns a promise which means that we don't need to use ```await``` as this is meant to "wait" for promises to be resolved before continuing with execution.